File: seed.py
# seed.py
from app import create_app, db
from app.models import User

app = create_app()

with app.app_context():
    db.create_all()
    
    # Criando os usuários iniciais já com o campo 'name'
    admin = User(name='Administrador do Sistema', username='admin', role='Admin')
    admin.set_password('admin123')
    
    operador = User(name='Operador BAMRJ', username='operador', role='Operador')
    operador.set_password('bamrj123')

    db.session.add_all([admin, operador])
    db.session.commit()
    print("[SUCESSO] Base do Sistema iniciada com sucesso!")

File: seed.php
<?php
require_once 'app/core/Database.php';
require_once 'app/models/User.php';

use App\Core\Database;
use App\Models\User;

// Simulação de variáveis de ambiente para execução via CLI local (opcional)
// No Railway, o Database.php já lerá as variáveis reais.

try {
    $db = Database::getConnection();
    
    $username = 'admin';
    $password = 'admin123'; // Senha padrão militar para o primeiro acesso
    $hash = User::hashPassword($password);

    $stmt = $db->prepare("INSERT INTO users (name, username, password_hash, role, must_change_password) 
                          VALUES (?, ?, ?, ?, ?)");
    
    $stmt->execute([
        'Administrador', 
        $username, 
        $hash, 
        'Admin', 
        false // O Admin principal começa sem a trava por padrão
    ]);

    echo "Utilizador Admin criado com sucesso no Railway!\n";
} catch (Exception $e) {
    echo "Erro ao criar admin: " . $e->getMessage();
}